full-service/src/json_rpc/wallet_api.rs

```diff
@@ -0,0 +1,880 @@
+                        .map_err(|e| format!("{{\"error\": \"{:?}\"}}", e))?,
+                )
+                .map_err(|e| format!("{{\"error\": \"{:?}\"}}", e))?;
+            Response::get_block_object {
+                block,
+                block_contents,
+            }
+        }
+        Request::get_proofs { transaction_log_id } => Response::get_proofs {
+            proofs: service
+                .get_proofs(&transaction_log_id)
+                .map_err(|e| format!("{{\"error\": \"{:?}\"}}", e))?,
+        },
+        Request::verify_proof {
+            account_id,
+            txo_id,
+            proof,
+        } => {
+            let result = service
+                .verify_proof(&account_id, &txo_id, &proof)
+                .map_err(|e| format!("{{\"error\": \"{:?}\"}}", e))?;
+            Response::verify_proof { verified: result }
+        }
+    };
+    Ok(Json(result))
+}
+
+#[post("/wallet", format = "json", data = "<command>")]
+fn wallet_api(
+    state: rocket::State<WalletApiState>,
+    command: Json<Request>,
+) -> Result<Json<Response>, String> {
+    wallet_api_inner(state.service.clone(), command)
+}
+
+#[get("/wallet")]
+fn wallet_help() -> Result<String, String> {
+    let mut help_str = "Please use json data to choose wallet commands. For example, \n\ncurl -s localhost:9090/wallet -d '{\"method\": \"create_account\", \"params\": {\"name\": \"Alice\"}}' -X POST -H 'Content-type: application/json'\n\nAvailable commands are:\n\n".to_owned();
+    for e in Request::iter() {
+        help_str.push_str(&format!("{:?}\n\n", e));
+    }
+    Ok(help_str)
+}
+
+pub fn rocket(rocket_config: rocket::Config, state: WalletApiState) -> rocket::Rocket {
+    rocket::custom(rocket_config)
+        .mount("/", routes![wallet_api, wallet_help])
+        .manage(state)
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use crate::{json_rpc, service::MockWallet};
+    use mc_common::logger::{test_with_logger, Logger};
+    use rocket::{
+        http::{ContentType, Status},
+        local::Client,
+    };
+    use rocket_contrib::json::JsonValue;
+    use std::sync::atomic::{AtomicUsize, Ordering::SeqCst};
+
+    fn get_free_port() -> u16 {
+        static PORT_NR: AtomicUsize = AtomicUsize::new(0);
+        PORT_NR.fetch_add(1, SeqCst) as u16 + 30300
+    }
+
+    /// Submit a JSON-RPC request.
+    fn dispatch(client: &Client, body: JsonValue, _logger: &Logger) -> serde_json::Value {
+        let mut res = client
+            .post("/wallet")
+            .header(ContentType::JSON)
+            .body(body.to_string())
+            .dispatch();
+        assert_eq!(res.status(), Status::Ok);
+        let body = res.body().unwrap().into_string().unwrap();
+        // log::info!(logger, "Attempted dispatch got response {:?}", body);
+        let res: JsonValue = serde_json::from_str(&body).unwrap();
+        res.get("result").unwrap().clone()
+    }
+
+    // fn dispatch_expect_error(
```

Comment:
I don't think this should be commented out - are the expected errors not firing?

Reply:
I see - this was commented out because all the tests were commented out.